src/rdkafka_assignor.c

```diff
@@ -431,16 +447,30 @@ rd_kafka_assignor_add (rd_kafka_t *rk,
 		       rd_kafka_assignor_t **rkasp,
                        const char *protocol_type,
                        const char *protocol_name,
+                       int supported_protocols,
                        rd_kafka_resp_err_t (*assign_cb) (
                                rd_kafka_t *rk,
+                               const struct rd_kafka_assignor_s *rkas,
                                const char *member_id,
-                               const char *protocol_name,
                                const rd_kafka_metadata_t *metadata,
                                rd_kafka_group_member_t *members,
                                size_t member_cnt,
                                rd_kafka_assignor_topic_t **eligible_topics,
                                size_t eligible_topic_cnt,
                                char *errstr, size_t errstr_size, void *opaque),
+                       rd_kafkap_bytes_t *(*get_metadata_cb) (
+                               const struct rd_kafka_assignor_s *rkas,
+                               void *assignor_state,
+                               const rd_list_t *topics,
+                               const rd_kafka_topic_partition_list_t
+                               *owned_partitions),
+                       void (*on_assignment_cb) (
+                               const struct rd_kafka_assignor_s *rkas,
+                               void **assignor_state,
+                               const rd_kafka_topic_partition_list_t *assignment,
+                               const rd_kafkap_bytes_t *userdata,
+                               const rd_kafka_consumer_group_metadata_t *rkcgm),
+                       void (*destroy_state) (void *assignor_state),
```

Comment:
this should have the _cb suffix to be consistent with the other callbacks here.

Reply:
yep, also `rkas_destroy_state_cb`